From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Chris Rummel <crummel@microsoft.com>
Date: Wed, 26 Oct 2022 17:03:02 -0500
Subject: [PATCH] Target only net7.0 during source-build - Port of #75168 to
 main (#77052)

* Target only net7.0 during source-build
Port of #75168 to main.

* Target net7.0 in additional projects

* Update compatibility suppressions

* Only target net7.0 for source-build

* Disable package validation for source-build

* Revert compatibility suppression changes

* Use NetCoreAppCurrent instead of hardcoding net7.0

* Revert mono Apple TFM change

* Remove unnecessary LangVersion

* Address code review comments.

Co-authored-by: Logan Bussell <loganbussell@microsoft.com>

Backport: https://github.com/dotnet/runtime/pull/77052
---
 Directory.Build.props | 1 +
 eng/SourceBuild.props | 1 +
 2 files changed, 2 insertions(+)

diff --git a/Directory.Build.props b/Directory.Build.props
index b902a691666..35e506f1eba 100644
--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -53,6 +53,7 @@
     <NetCoreAppCurrentBrandName>.NET $(NetCoreAppCurrentVersion)</NetCoreAppCurrentBrandName>
     <NetCoreAppCurrent>net$(NetCoreAppCurrentVersion)</NetCoreAppCurrent>
     <NetCoreAppMinimum>net6.0</NetCoreAppMinimum>
+    <NetCoreAppMinimum Condition="'$(DotNetBuildFromSource)' == 'true'">$(NetCoreAppCurrent)</NetCoreAppMinimum>
 
     <NetCoreAppToolCurrentVersion>7.0</NetCoreAppToolCurrentVersion>
     <NetCoreAppToolCurrent>net$(NetCoreAppToolCurrentVersion)</NetCoreAppToolCurrent>
diff --git a/eng/SourceBuild.props b/eng/SourceBuild.props
index ebf8e683ed0..1dfe97fe1ca 100644
--- a/eng/SourceBuild.props
+++ b/eng/SourceBuild.props
@@ -48,6 +48,7 @@
       <InnerBuildArgs>$(InnerBuildArgs) /p:BuildDebPackage=false</InnerBuildArgs>
       <InnerBuildArgs>$(InnerBuildArgs) /p:EnableNgenOptimization=false</InnerBuildArgs>
       <InnerBuildArgs>$(InnerBuildArgs) /p:AdditionalRuntimeIdentifierParent=$(BaseOS)</InnerBuildArgs>
+      <InnerBuildArgs>$(InnerBuildArgs) /p:EnablePackageValidation=false</InnerBuildArgs>
     </PropertyGroup>
   </Target>
 
